### PR TITLE
Prefix test - Different kind of load test to trigger prefix test bugs.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: backends/v3/src/radix.rs [hunk 1]
```diff
@@ -70,9 +70,15 @@ impl Allocator for RadixAllocator {
...
-                .cache_blocks
-                .find(prefill_tokens.as_slice(), &mut blocks);
+            let node_id = self.cache_blocks.find(
+                // XXX This is super important we cannot match an entire prefix
+                // otherwise input_ids is empty and the shard code cannot handle that.
+                &prefill_tokens.as_slice()[..prefill_tokens.len().saturating_sub(1)],
+                &mut blocks,
+            );
+            // Even if this allocation fails below, we need to increase he
+            // refcount to ensure that the prefix that was found is not evicted.
```
Comment (danieldk) [inline backends/v3/src/radix.rs:80]: This comment is already (and should be) below.

**Review by danieldk — APPROVED** [review]
One small comment for polish. LGTM otherwise.

**Comment by danieldk** [comment]
CI looks bugged though?

**Comment by Narsil** [comment]
> CI looks bugged though?

Unfortunately no it's not bugged the issue was actually happening on docker (flashinfer 1.4.0 vs 1.5.0 at least)

**Review by OlivierDehaene — APPROVED** [review]

**Pull request merged**